omnigibson/prims/material_prim.py

```diff
@@ -231,6 +231,20 @@ def set_input(self, inp, val):
             val (any): Value to set for the input. This should be the valid type for that attribute.
         """
         # Make sure the input exists first, so we avoid segfaults with "invalid null prim"
+        if inp not in self.shader_input_names:
+            input_type = None
+            if isinstance(val, str):
+                input_type = lazy.pxr.Sdf.ValueTypeNames.String
+            elif isinstance(val, int):
+                input_type = lazy.pxr.Sdf.ValueTypeNames.Int
+            elif isinstance(val, float):
+                input_type = lazy.pxr.Sdf.ValueTypeNames.Float
+            elif isinstance(val, bool):
+                input_type = lazy.pxr.Sdf.ValueTypeNames.Bool
+            else:
+                raise ValueError(f"Invalid input type {type(val)} for input {inp}")
```

Comment:
I remember Hang has a helper function that does this. If not, maybe put this into a helper function in one of the utils files?

Reply:
He does in an unmerged PR (the 4.2 and 4.5 branches), yes - he'll merge that PR in a bit so this can quickly get removed when he does that. I can't put it here because it uses the new API for checking what inputs are available which is 4.2+.